tests/integration/test_kafka_edge_cases.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2023 Aineko Authors
+# SPDX-License-Identifier: Apache-2.0
+"""Tests edge cases to do with kafka clusters."""
+
+import time
+from typing import Optional
+
+import pytest
+import ray
+
+from aineko import AbstractNode, DatasetConsumer, Runner
+
+
+class ConsumerNode(AbstractNode):
+    """Node that consumes message using different consume methods."""
+
+    def _execute(self, params: Optional[dict] = None) -> None:
+        """Consumes message."""
+        self.consumers["messages"].consume(how="next")
+        self.consumers["messages"].consume(how="last", timeout=1)
+        self.producers["test_result"].produce("OK")
+        self.producers["test_result"].produce("END")
+        time.sleep(1)
+        self.activate_poison_pill()
+        return False
+
+
+@pytest.mark.integration
+def test_consume_empty_datasets():
+    """Integration test that checks that empty datasets do not cause errors.
+
+    If a dataset is empty, dataset consumer methods should not error out.
+    """
+    runner = Runner(
+        pipeline_config_file="tests/conf/integration_test_empty_dataset.yml",
+    )
+    try:
+        runner.run()
+    except ray.exceptions.RayActorError:
+        consumer = DatasetConsumer(
+            dataset_name="test_result",
+            node_name="consumer",
+            pipeline_name="integration_test_kafka_edge_cases",
+            dataset_config={},
+            has_pipeline_prefix=True,
+        )
+        count_messages = consumer.consume_all(end_message="END")
+        assert count_messages[0]["message"] == "OK"
```

Comment:
At first glance I was confused by what you're doing here. I don't think it's very common to use a try-except like that. I would suggest refactoring it so it's easier to understand to everyone.

```suggestion
    try:
        runner.run()
    except ray.exceptions.RayActorError:
        # This is expected because we activated the poison pill
        pass

    consumer = DatasetConsumer(
        dataset_name="test_result",
        node_name="consumer",
        pipeline_name="integration_test_kafka_edge_cases",
        dataset_config={},
        has_pipeline_prefix=True,
    )
    count_messages = consumer.consume_all(end_message="END")
    assert count_messages[0]["message"] == "OK"
```

Reply:
Also fwiw this test passes on the current develop, so without your fix. I assume this is because the dataset wasn't purged locally since I first ran the test on this branch and then on develop. If that's the case, our tests would need to do some clean-up before and after running. What do you think?

Reply:
I agree with Erwin on not putting logic inside an `except` block. It should only be used for error handling.